msal/src/main/java/com/microsoft/identity/client/statemachine/states/BaseStates.kt

```diff
@@ -0,0 +1,289 @@
+
+            val params = CommandParametersAdapter.createRemoveAccountCommandParameters(
+                config,
+                config.oAuth2TokenCache,
+                requestAccountRecord
+            )
+
+            val removeCurrentAccountCommandParameters = RemoveCurrentAccountCommand(
+                params,
+                LocalMSALController(),
+                object : CommandCallback<Boolean?, BaseException?> {
+                    override fun onError(error: BaseException?) {
+                        // Do nothing, handled by CommandDispatcher.submitSilentReturningFuture()
+                    }
+
+                    override fun onTaskCompleted(result: Boolean?) {
+                        // Do nothing, handled by CommandDispatcher.submitSilentReturningFuture()
+                    }
+
+                    override fun onCancel() {
+                        // Do nothing
+                    }
+                },
+                PublicApiId.NATIVE_AUTH_ACCOUNT_SIGN_OUT
+            )
+
+            val result = CommandDispatcher.submitSilentReturningFuture(removeCurrentAccountCommandParameters)
+                .get().result as Boolean
+
+            return@withContext if (result) {
+                SignOutResult.Complete
+            } else {
+                Logger.error(
+                    TAG,
+                    "Unexpected error during signOut.",
+                    null
+                )
+                throw MsalClientException(
+                    MsalClientException.UNKNOWN_ERROR,
+                    "Unexpected error during signOut."
+                )
+            }
+        }
+    }
+
+    /**
+     * Gets the current account.
+     *
+     * @return account [com.microsoft.identity.client.IAccount].
+     */
+    fun getAccount(): IAccount {
+        return account
+    }
+
+    /**
+     * Gets the current account's ID token (if present).
+     *
+     * @return idToken [String].
+     */
+    fun getIdToken(): String? {
+        return account.idToken
+    }
+
+    /**
+     * Gets the claims associated with the current account.
+     *
+     * @return A Map of claims.
+     */
+    fun getClaims(): Map<String, *>? {
+        return account.claims
+    }
+
+    interface GetAccessTokenCallback : Callback<IAuthenticationResult?>
+
+    /**
+     * Retrieves the access token for the currently signed in account from the cache.
+     * If the access token is expired, it will be attempted to be refreshed using the refresh token that's stored in the cache;
+     * callback variant.
+     *
+     * @return [com.microsoft.identity.client.IAuthenticationResult] If successful.
+     * @throws [MsalClientException] If the the account doesn't exist in the cache.
+     * @throws [ServiceException] If the refresh token doesn't exist in the cache/is expired, or the refreshing fails.
+     */
+    fun getAccessToken(forceRefresh: Boolean = false, callback: GetAccessTokenCallback) {
```

Comment:
You can get access token from a state where you never signed in?

Reply:
Same reply as for the other comment: this method is part of `AccountResult`, which is a result/state that is returned when you're signed in.

Reply:
I have moved the `AccountResult` to its own file so its clear that BaseStates interface doesn't have the `signOut` and  `getAccessToken` method.